src/overlays/actors/ovl_En_Mm2/z_en_mm2.c

```diff
@@ -31,16 +30,57 @@ const ActorInit En_Mm2_InitVars = {
     (ActorFunc)EnMm2_Draw,
 };
 
-#endif
+#include "overlays/ovl_En_Mm2/ovl_En_Mm2.c"
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Mm2/EnMm2_Init.s")
+void EnMm2_Init(Actor* thisx, GlobalContext* globalCtx) {
+    EnMm2* this = THIS;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Mm2/EnMm2_Destroy.s")
+    Actor_SetScale(&this->actor, 0.015f);
+    this->actionFunc = EnMm2_WaitForRead;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Mm2/func_809A2080.s")
+void EnMm2_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Mm2/func_809A20FC.s")
+// Action function whilst Link is reading the letter.
+void EnMm2_Reading(EnMm2* this, GlobalContext* globalCtx) {
+    u8 temp_v0;
```

Comment:
```suggestion
   u8 talkState = func_80152498(&globalCtx->msgCtx);
```
Would moving things to one line work? (The name is just a suggestion)